src/commands/parser/parser.ts

```diff
@@ -118,29 +118,98 @@ function processCommand(lexer: Lexer): Command {
-      // try to match the typeToDestroy token value to a known type token
-      const lowerMaybeTypeToDestroy = typeToDestroyToken.value.toLowerCase()
+    // destroy takes an optional radius
+    let typeToDestroy: ToDestroy | undefined
+    let coalition: Coalition | undefined
+    let radius: number | undefined
 
-      if (/^unit/.test(lowerMaybeTypeToDestroy) === true) {
-        return {
-          type: CommandType.Destroy,
-          toDestroy: ToDestroy.Unit,
-        }
+    const parseParts = (): void => {
+      const nextToken = lexer.nextToken()
+
+      if (TokenKind.EOF === nextToken.kind) {
+        return
       }
-      if (/^spawner/.test(lowerMaybeTypeToDestroy) === true) {
-        return {
-          type: CommandType.Destroy,
-          toDestroy: ToDestroy.Spawner,
+
+      // if string or word, assume it's typeToDestroy
+      if (
+        TokenKind.String === nextToken.kind ||
+        TokenKind.Word === nextToken.kind
+      ) {
+        const lowerValue = nextToken.value.toLowerCase()
+
+        // try to match string token value to a known type or coalition token
+        if (/^unit/.test(lowerValue) === true) {
+          typeToDestroy = ToDestroy.Unit
+          return parseParts()
+        }
+        if (/^spawner/.test(lowerValue) === true) {
+          typeToDestroy = ToDestroy.Spawner
+          return parseParts()
+        }
+
+        // check for coalition
+        if (lowerValue === 'red') {
+          coalition = Coalition.COALITION_RED
+          return parseParts()
+        }
+        if (lowerValue === 'blue') {
+          coalition = Coalition.COALITION_BLUE
+          return parseParts()
+        }
+        if (lowerValue === 'all') {
+          coalition = Coalition.COALITION_ALL
+          return parseParts()
+        }
+
+        // takes rad(ius) as keyword and expects value in meters in next token
+        if (/^rad/.test(lowerValue) === true) {
+          const radiusValueToken = lexer.nextToken()
+
+          if (TokenKind.Number !== radiusValueToken.kind) {
+            throw new Error('expected number token following radius keyword')
+          }
+
+          radius = radiusValueToken.value
+
+          return parseParts()
         }
+        return parseParts()
+      }
+
+      // if number, assume it's radius
+      if (TokenKind.Number === nextToken.kind) {
+        radius = nextToken.value
+        return parseParts()
+      }
+    }
+
+    parseParts()
+
+    if (typeToDestroy != undefined && radius != undefined) {
+      return {
+        type: CommandType.Destroy,
+        toDestroy: typeToDestroy,
+        radius: radius,
+        coalition: coalition,
+      }
+    }
+    if (typeToDestroy != undefined) {
+      return {
+        type: CommandType.Destroy,
+        toDestroy: typeToDestroy,
+        coalition: coalition,
+      }
+    }
+    if (radius != undefined) {
+      return {
+        type: CommandType.Destroy,
+        radius: radius,
+        coalition: coalition,
```

Comment:
this complexity feels like something we should address 🤔 I feel like it makes it harder to write code here. with so much boilerplate

being commands, I lean towards being explicit. Separating the logic of "what did the user put in" (the command knows about radius, but the user input nothing for radius) vs "what does this module want to do with that data"

where a coalition is explicitly required in the command?? but the user can also input nothing for coalition.

The "shape" of a `DestroyCommand` should, imo, always have every property. as that's the command. a command doesn't lack a radius just because the user didn't input one, it has explicit knowledge ( the property existing with value `undefined`) that the user intentionally didn't provide the value.

I haven't entirely convinced myself here yet, and welcome other input.

for the sake of this pr, we should ship what you have here.

Reply:
changed it, because it makes sense to me